Sources/StreamChatUI/ViewContainerBuilder/ViewContainerBuilder.swift

```diff
@@ -0,0 +1,270 @@
+    stack.alignment = alignment
+    stack.spacing = spacing
+    return stack
+}
+
+/// A flexible space that expands along the major axis of its containing stack layout.
+public func Spacer() -> UIView {
+    let view = UIView()
+    view.translatesAutoresizingMaskIntoConstraints = false
+    return view
+}
+
+// MARK: - Layout & Constraints Builders
+
+public extension UIView {
+    /// Syntax sugar to be able to add additional layout changes in place when building layouts with the `@ViewContainerBuilder`.
+    /// Example:
+    /// ```
+    /// replyTimestampLabel.layout {
+    ///   $0.setContentCompressionResistancePriority(.required, for: .horizontal)
+    ///
+    ///   NSLayoutConstraint.activate([
+    ///     $0.heightAnchor.pin(equalToConstant: 15)
+    ///     $0.widthAnchor.pin(equalToConstant: 15)
+    ///   ])
+    /// }
+    /// ```
+    @discardableResult
+    func layout(_ block: (Self) -> Void) -> Self {
+        block(self)
+        return self
+    }
+
+    /// Syntax sugar to be able to set view constraints in place when building layouts with the `@ViewContainerBuilder`.
+    /// The constraints are automatically activated.
+    ///
+    /// Example:
+    /// ```
+    /// threadIconView.constraints {
+    ///   $0.heightAnchor.pin(equalToConstant: 15)
+    ///   $0.widthAnchor.pin(equalToConstant: 15)
+    /// }
+    /// ```
+    @discardableResult
+    func constraints(@ViewContainerBuilder block: (Self) -> [NSLayoutConstraint]) -> Self {
+        NSLayoutConstraint.activate(block(self))
+        return self
+    }
+}
+
+// MARK: - UIStackView.views {} - Helper to replace the subviews
+
+public extension UIStackView {
+    /// Result builder to allow replacing views of a container.
+    /// This is useful when containers have a reference.
+    ///
+    /// /// Example:
+    /// ```
+    /// container.views {
+    ///     threadIconView
+    ///     replyTimestampLabel
+    /// }
+    /// ```
+    @discardableResult
+    func views(
+        @ViewContainerBuilder _ subviews: () -> [UIView]
+    ) -> Self {
+        removeAllArrangedSubviews()
+        subviews().forEach { addArrangedSubview($0) }
+        return self
+    }
+}
+
+// MARK: - UIStackView.embed() - Helper to add container to parent view
+
+public extension UIStackView {
+    /// Embeds the container to the given view.
+    @discardableResult
+    func embed(in view: UIView) -> UIStackView {
+        view.addSubview(self)
+        pin(to: view)
+        return self
+    }
+
+    /// Embeds the container to the given view with insets.
+    @discardableResult
+    func embed(in view: UIView, insets: NSDirectionalEdgeInsets) -> UIStackView {
+        view.embed(self, insets: insets)
+        return self
+    }
+
+    /// Embeds the container to the given view respecting the layout margins guide.
+    /// The margins can be customised by changing the `directionalLayoutMargins`.
+    @discardableResult
+    func embedToMargins(in view: UIView) -> UIStackView {
```

Comment:
I would prefer that to be more generic. One method where you can pass either a view or a layout guide

Reply:
Yeah I wanted to make this API a little more beginner friendly, but maybe better to stick to Apple's API 🤔 I'll see if I can improve this 👍

Reply:
I tried this but it makes it less simple, because you would still need to provide the view, so that we add the container to the parent view. So something like this would need to be done, which does not look as good IMO:

```swift
func embed(in view: UIView, to margins: UILayoutGuide) -> UIStackView
```

Reply:
This would probably look like this:

```swift
func embed(in view: UIView, margins: UILayoutGuide) -> UIStackView
```

which would be consistent with the existing:
```swift
func embed(in view: UIView, insets: NSDirectionalEdgeInsets) -> UIStackView
```

WDYT?